src/shared/ask-password-api.c

```diff
@@ -113,6 +113,9 @@ static int add_to_keyring(const char *keyname, AskPasswordFlags flags, char **pa
         if (r < 0)
                 return r;
 
+        n = LESS_BY(n, 1); /* chop off the final NUL byte. We do this because we want to use the separator
```

Comment:
It seems `1` needs to be casted with `(size_t)`.
```
In file included from ../src/basic/macro.h:13,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/ask-password-api.c:21:
../src/shared/ask-password-api.c: In function ‘add_to_keyring’:
../src/fundamental/macro-fundamental.h:216:31: error: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Werror=sign-compare]
  216 |                 UNIQ_T(A, aq) > UNIQ_T(B, bq) ? UNIQ_T(A, aq) - UNIQ_T(B, bq) : 0; \
      |                               ^
../src/fundamental/macro-fundamental.h:211:23: note: in expansion of macro ‘__LESS_BY’
  211 | #define LESS_BY(a, b) __LESS_BY(UNIQ, (a), UNIQ, (b))
      |                       ^~~~~~~~~
../src/shared/ask-password-api.c:116:13: note: in expansion of macro ‘LESS_BY’
  116 |         n = LESS_BY(n, 1); /* chop off the final NUL byte. We do this because we want to use the separator
      |             ^~~~~~~
cc1: all warnings being treated as errors
```